src/clj/rems/workflow/dynamic.clj

```diff
@@ -156,8 +156,8 @@
 
 (defn- validation-error
   [injections application-id]
-  (when (not ((:valid-form-inputs? injections) application-id))
-    {:errors [:form-not-valid]}))
+  (when-let [errors ((:validate-form injections) application-id)]
+    {:errors errors}))
```

Comment:
should the errors be tagged as validation errors? perhaps something like
```
{:errors [[:validation errors]]}
```
or even
```
{:errors (mapv (fn [e] [:validation e]) errors)}
```